app/bin/databricks_common_utils.py

```diff
@@ -202,3 +253,68 @@ def format_to_json_parameters(params):
         )
 
     return output_json
+
+
+def get_aad_access_token(session_key, user_agent, proxy_settings=None,
```

Comment:
You can avoid too many calls to AAD API if you will check `expire` field into a consideration